node/router/src/heartbeat.rs

```diff
@@ -117,32 +154,15 @@ pub trait Heartbeat<N: Network>: Outbound<N> {
             return;
         }
 
-        // Retrieve the trusted peers.
-        let trusted = self.router().trusted_peers();
-        // Retrieve the bootstrap peers.
-        let bootstrap = self.router().bootstrap_peers();
-
-        // Find the oldest connected peer, that is neither trusted nor a bootstrap peer.
-        let oldest_peer = self
-            .router()
-            .get_connected_peers()
-            .iter()
-            .filter(|peer| !trusted.contains(&peer.ip()) && !bootstrap.contains(&peer.ip()))
-            .filter(|peer| !self.router().cache.contains_inbound_block_request(&peer.ip())) // Skip if the peer is syncing.
-            .filter(|peer| self.is_block_synced() || self.router().cache.num_outbound_block_requests(&peer.ip()) == 0) // Skip if you are syncing from this peer.
-            .min_by_key(|peer| peer.last_seen())
-            .map(|peer| peer.ip());
-
         // Disconnect from the oldest connected peer, if one exists.
-        if let Some(oldest) = oldest_peer {
+        if let Some(oldest) = self.get_removable_peers().pop().map(|peer| peer.ip()) {
```

Comment:
I might be misreading this but if validators are sorted to the end of the vector, is `pop` what we want here?

Reply:
You are right. It should be fixed now.